pkg/bloomcompactor/spec.go

```diff
@@ -313,6 +311,12 @@ func (b *batchedLoader) Next() bool {
 	}
 
 	b.batch, b.err = next.fetcher.FetchChunks(b.ctx, toFetch)
+	return b.prepNext()
```

Comment:
FetchChunks may return an error, then `len(b.batch)` is 0, which will cause an out of bounds error in `prepNext()`.

Reply:
Good catch, will fix. Thanks